Lib/os.py

```diff
@@ -847,6 +848,14 @@ def _spawnvef(mode, file, args, env, func):
             raise TypeError('argv must be a tuple or a list')
         if not args or not args[0]:
             raise ValueError('argv first element cannot be empty')
+
+        sig = inspect.signature(func)
```

Comment:
I don't think that it's appropriate to use inspect here. I would prefer to see the check in the caller function instead. Add an helper function if you want to factorize the code.

Reply:
@vstinner So you mean this validation should be placed on function which call this `_spawnvef` function, right?

Reply:
Move the test into spawnve() and spawnvpe().